### PR TITLE
Witek/add rewards

### PR DESCRIPTION
**Pull request by piowit**

### CODE REVIEW

File: app/models/reward.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class Reward < ApplicationRecord
+  validates :price, numericality: { greater_than: 0 }
```
Comment (ciejj) [inline app/models/reward.rb]: According to the Acceptance criteria:
_Price must be greater than or equal to 1_

File: app/models/employee.rb [hunk 2]
```diff
@@ -8,4 +8,5 @@ class Employee < ApplicationRecord
 
   has_many :given_kudos, class_name: 'Kudo', foreign_key: 'giver_id', dependent: :destroy, inverse_of: :giver_of_kudo
   has_many :received_kudos, class_name: 'Kudo', foreign_key: 'receiver_id', dependent: :destroy, inverse_of: :receiver_of_kudo
+  # has_many :rewards, dependent: :destroy
```
Comment (ciejj) [inline app/models/employee.rb]: In general leaving comments in code is not a good practice

File: app/models/reward.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+  validates :title, :description, :price, presence: true
+  # has_one :employee, dependent: :destroy
```
Comment (ciejj) [inline app/models/reward.rb]: In general leaving comments in code is not a good practice

[edit] bonus question - would this association work in current (commented out) configuration?

File: spec/models/reward_spec.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Reward, type: :model do
+  subject(:reward) { described_class.new(title: 'test', description: 'test', price: 1) }
```
Comment (ciejj) [inline spec/models/reward_spec.rb]: I would suggest using FactoryBot with `build`

```suggestion
  let(:reward) { build(:reward, title: 'test', description: 'test', price: 1) }
```

File: spec/models/reward_spec.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+  it 'is not valid with price lower than 0' do
+    reward.price = -1
+    expect(reward).not_to be_valid
+  end
+
+  it 'is not valid with price equal 0' do
+    reward.price = 0
+    expect(reward).not_to be_valid
+  end
```
Comment (ciejj) [inline spec/models/reward_spec.rb]: I think you can merge those two examples into one, since in both cases you are checking value which is lower than 1.

File: spec/factories/rewards.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :reward do
+    sequence(:title) { |i| "Title#{i}" }
+    sequence(:description) { |i| "Description#{i}" }
+    sequence(:price) { |i| i }
+    title { 'MyString' }
+    description { 'MyText' }
+    price { '9.99' }
```
Comment (ciejj) [inline spec/factories/rewards.rb]: I  think this is unnecessary - sequence helpers are covering title, description, and price

**Review by ciejj — CHANGES_REQUESTED** [review]
Good job (and blazing fast :), but in my understanding one o the Acceptance Criteria is not met - other comments are pure cosmetics 👍

File: app/models/reward.rb (hunk 1 above)
Comment (piowit) [reply app/models/reward.rb]: i had fixed in my mind that is should be grater then zero, idk why :)

File: app/models/employee.rb (hunk 2 above)
Comment (piowit) [reply app/models/employee.rb]: oh i wanted to do too much, i started doing association, and later i read the task again and commented it out for later removal, and forgot to do this :/

File: app/models/reward.rb (hunk 3 above)
Comment (piowit) [reply app/models/reward.rb]: i think not, because on one side of association we should have has_one/many/etc while on the other side we need to have belongs_to

File: spec/factories/rewards.rb (hunk 6 above)
Comment (piowit) [reply spec/factories/rewards.rb]: leftover, forgot to delete

**Review by ciejj — APPROVED** [review]
That was fast :) good job! 👍

**Comment by piowit** [comment]
there's no time to waste :)

File: spec/factories/rewards.rb (hunk 6 above)
Comment (ciejj) [reply spec/factories/rewards.rb]: Actually I have no clue how would the model created with that factory look like - I mean would it have attributes from sequence or regular assignment? Maybe it is worth checking and sharing with group as TIL?

Comment (piowit) [reply spec/factories/rewards.rb]: it was generated by using `rails g model reward title:string description:text price:decimal`, and after this we are getting model, migration, empty test spec, and factory that looks like these deleted lines :)

Comment (ciejj) [reply spec/factories/rewards.rb]: I mean that you have all attributes defined twice - normal way (generator) and with sequence.

So when you create an object with that factory which attributes will it have?

Comment (piowit) [reply spec/factories/rewards.rb]: ohh thats because i was editing this factory by pushing down old definitions and creating sequences to make sure everything is correctly covered, and then i forgot to delete this unnecessary lines

Comment (ciejj) [reply spec/factories/rewards.rb]: ok, let me rephrase the question

we have a (simplified) factory like this:

```
factory :reward do
    sequence(:title) { |i| "Foo_#{i}" }
    title { 'Bar' }
end
```

then if I create the reward object with that factory:

`let!(new_reward) {create(:reward)}`

what value will `new_reward.title` have?

As previously mentioned - I do not know, but maybe you want to investigate it, and share with the group :)

**Review by oskarlakner — APPROVED** [review]
